hphp/runtime/vm/jit/service-requests.cpp

```diff
@@ -254,42 +266,35 @@ FPInvOffset extract_spoff(TCA stub) {
     }
 
     case Arch::ARM: {
-      struct Decoder : public vixl::Decoder {
-        void VisitAddSubImmediate(vixl::Instruction* inst) {
-          // For immediate operands, shift can be '0' or '12'
-          int64_t immed =
-            inst->ImmAddSub() << ((inst->ShiftAddSub() == 1) ? 12 : 0);
-          switch (inst->Mask(vixl::AddSubOpMask)) {
-            case vixl::ADD: offset = immed; break;
-            case vixl::SUB: offset = -immed; break;
-            default: break;
-          }
-        }
-        void VisitMoveWideImmediate(vixl::Instruction* inst) {
-          // For wide moves, shift can be 0, 16, 32 or 64
-          int64_t immed = safe_cast<int64_t>(
-            inst->ImmMoveWide() << (inst->ShiftMoveWide() << 4));
-          switch (inst->Mask(vixl::MoveWideImmediateMask)) {
-            case vixl::MOVN_w:
-            case vixl::MOVN_x:
-              immed = safe_cast<int64_t>(~immed);
-              break;
-          }
-          offset = immed;
-        }
-        folly::Optional<int32_t> offset;
-      };
-      Decoder decoder;
-      decoder.Decode((vixl::Instruction*)(stub));
-
-      // 'lea' becomes
-      //   a. 'add dst, base, #imm' or
-      //   b. 'mov r, #imm'
-      //      'add dst, base, r'
-      // FIXME: Return '0' if vasm optimizes 'lea' to 'mov'
-      if (!decoder.offset) return FPInvOffset{0};
-      always_assert(decoder.offset && (*decoder.offset % sizeof(Cell)) == 0);
-      return FPInvOffset{-(*decoder.offset / int32_t{sizeof(Cell)})};
+      auto instr = reinterpret_cast<vixl::Instruction*>(stub);
+
+      if (instr->IsAddSubImmediate()) {
+        auto const offBytes = safe_cast<int32_t>(instr->ImmAddSub());
+        always_assert((offBytes % sizeof(Cell)) == 0);
+
+        if (instr->Mask(vixl::AddSubImmediateMask) == vixl::SUB_w_imm ||
+            instr->Mask(vixl::AddSubImmediateMask) == vixl::SUB_x_imm)
+          return FPInvOffset{offBytes / int32_t{sizeof(Cell)}};
```

Comment:
Just to be a little more clear - please surround the return statement with braces so it's not a single hanging statement.